### ISSUE TITLE
Make Exception a KnownType

### ISSUE THREAD

**Issue by axunonb** [YAXLib/YAXLib#197]
Even with a `MaxRecursion` of 10, serializatiion output is too heavy and serialization takes too long.
Add `Exception` to `KnownTypes` with reduced serialization levels and/or members.

**Comment by axunonb** [comment]
https://github.com/YAXLib/YAXLib/pull/152 never merged into master?

**Comment by axunonb** [comment]
Memory usage is very high, even with low setting for `MaxRecursion`:

```
MaxRecursion = 10
|             Method |  N |     Mean | Error |        Gen 0 |       Gen 1 |      Gen 2 | Allocated |
|------------------- |--- |---------:|------:|-------------:|------------:|-----------:|----------:|
| SerializeException |  1 | 177.7 ms |    NA |    3000.0000 |   1000.0000 |          - |     30 MB |
| SerializeException | 10 | 994.5 ms |    NA |   40000.0000 |  11000.0000 |  4000.0000 |    296 MB |

MaxRecursion = 30
|             Method |  N |     Mean | Error |        Gen 0 |       Gen 1 |      Gen 2 | Allocated |
|------------------- |--- |---------:|------:|-------------:|------------:|-----------:|----------:|
| SerializeException |  1 |  3.496 s |    NA |  126000.0000 |  17000.0000 |  4000.0000 |    986 MB |
| SerializeException | 10 | 32.475 s |    NA | 1257000.0000 | 168000.0000 | 29000.0000 |  9,857 MB |
```
Code (as used in the unit tests):
```CSharp
try
{
    throw new ArgumentOutOfRangeException("index",
        new InvalidOperationException("Inner exception 1",
            new Exception("Inner Exception 2")));
}
catch (Exception ex)
{
    var ser = new YAXSerializer<Exception>(new SerializerOptions { MaxRecursion = 30 });
    _ = ser.SerializeToXDocument(ex);
}
```

**Comment by axunonb** [comment]
@304NotModified  I'm just reviewing the `KnownTypeTests` and came across
https://github.com/YAXLib/YAXLib/blob/5a37f7fe33ed66ffe72f25205f7bdd32b44060eb/YAXLibTests/KnownTypeTests.cs#L27-L41

In Line 36, finally `RuntimeTypeDynamicKnownType.Serialize(...)` is invoked instead of `TypeKnownType.Serialize(...)`,
as `typeExample` is a runtime type. Could you give a hint for fixing this (if possible at all)...

**Comment by axunonb** [comment]
Found the bug: `GetType()` was called for a `Type`.

**Comment by 304NotModified** [comment]
> #152 never merged into master?

Whoops indeed, also not in version/4,0 it seems

**Comment by axunonb** [comment]
Resolved with release [v4.0.0](https://github.com/YAXLib/YAXLib/releases/tag/v4.0.0)
